homeassistant/config_entries.py

```diff
@@ -1630,9 +1630,9 @@ def _index_entry(self, entry: ConfigEntry) -> None:
                     report_issue,
                 )
 
-            self._domain_unique_id_index.setdefault(entry.domain, {})[
-                unique_id_hash
-            ] = entry
+            self._domain_unique_id_index.setdefault(entry.domain, {}).setdefault(
```

Comment:
Side note: Could we initialize this attribute with defaultdicts, one with dict and one with list as default. I've seen Nick optimize other places replacing use of `setdefault` with that.